### ISSUE TITLE
Wrong __new__ method in the collections.UserString

### ISSUE THREAD

**Issue by pomponchik** [python/cpython#103155]
# Bug report

Hello!

I ran into a problem in trying to implement my own class inherited from [UserString](https://docs.python.org/3/library/collections.html#collections.UserString) and `str`. Let me demonstrate a piece of code:

```python
from collections import UserString


class MyString(UserString, str):
    def __init__(self, sec):
        self.sec = sec

    def __new__(cls, *args, **kwargs):
        return str.__new__(cls)

    @property
    def data(self):
        return self.sec


assert MyString('lol kek') not in 'kek'
```

The result:

```
Traceback (most recent call last):
  File "/Users/pomponchik/Desktop/Projects/lazy_fstrings/test.py", line 16, in <module>
    assert MyString('lol kek') not in 'kek'
AssertionError
```

In this situation, the check does not pass. The string "kek" is sure to include "lol kek", although this is obviously impossible since "lol kek" is longer than just "kek".

It seems that this problem is easily fixed by simply removing your override of the magic method `__new__()`.  Let's try it:

```python
from collections import UserString


class MyString(UserString, str):
    def __init__(self, sec):
        self.sec = sec

    @property
    def data(self):
        return self.sec


assert MyString('lol kek') not in 'kek'
```

It works! But unfortunately, this makes it impossible to use my class with more than one argument during initialization. I think this behavior is a bug, because I don't see any reason why initialization with more than one argument by overriding the magic methods `__new__` and `__init__` should be prohibited.

As far as I understand, `__new__` in this case has some hidden behavior that affects the behavior of the resulting object when checking for occurrence. It looks like it shouldn't be there.

# Environment

- CPython versions tested on: 3.9.2
- Operating system and architecture: MacOS Monterey (12.2.1), MacBook Air (M1, 2020)

**Comment by pomponchik** [comment]
I would also like to pay special attention to the fact that only "not in" checks are broken in this case.

**Comment by Eclips4** [comment]
I'm not sure that problem in `UserString`.
```
class Idk(str):
    def __new__(cls, *args, **kwargs):
        return str.__new__(cls)


assert Idk() not in 'kek'
```
> AssertionError

**Comment by Eclips4** [comment]
I'm have more researched this problem and make the conclusion:
In `__new__` you create a empty string. 
After that, your expression looks like `"" in "ANY_STRING"` which always be `True`.
So.. I don't see a problem there.

**Comment by gaogaotiantian** [comment]
If you do

```python
from collections import UserString
assert UserString('lol kek') not in 'kek'
```

You'll get an error, complaining that ``'in <string>' requires string as left operand, not UserString``. The only reason you can do ``assert MyString('lol kek') not in 'kek'`` is that `MyString` is also a `str`. So, when doing `in` operation, your `MyString` is treated as a `str`, which won't access `self.data` for the operation. It's using the internal data of `str` object.

The reason removing the ``__new__`` method works is that you'll properly created a valid string with the string you provided. You can also achieve that by doing

```python
    def __new__(cls, *args, **kwargs):
        return str.__new__(cls, *args)
```

So, in short, the `in` operation you are trying to do has nothing to do with `UserString` class, therefore all the `self.sec` and `self.data` won't affect it at all.
